docs/components/modal.md

```diff
@@ -1,348 +1,752 @@
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+>
+  <template #footer>
+    All prices are in USD.
+  </template>
+</KModal>
+
+```html
+<KModal
+  title="Title"
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+>
+  <template #footer>
+    All prices are in USD.
+  </template>
+</KModal>
+```
+
+### footer-actions
+
+Use this slot should you need to provide cusom buttons in modal footer. Ommited when [`footer` slot](#footer) is used.
+
+<KButton @click="modal19Visible = true">Modal</KButton>
+<KModal
+  title="Title"
+  :visible="modal19Visible"
+  @cancel="closeAllModals"
+>
+  <template #footer-actions>
+    <KButton appearance="tertiary" @click="closeAllModals">
+      <BackIcon />
+      Back
+    </KButton>
+    <KButton @click="closeAllModals">
+     Proceed
+     <ForwardIcon />
+    </KButton>
+  </template>
+</KModal>
+
+```vue
+<template>
   <KModal
-    :is-visible="slottedIsOpen3"
-    title="Look at my slots!"
-    content="You know you like these slots."
-    dismiss-button-theme="dark"
-    @canceled="slottedIsOpen3 = false"
+    :visible="modalVisible"
+    title="Modal"
+    @cancel="handleModalClose"
   >
-    <template v-slot:header-image>
-      <div class="slot-image-content">
-        <h4>I'm slotted baby!</h4>
-      </div>
+    <template #footer-actions>
+      <KButton
+        appearance="tertiary"
+        @click="handleModalClose">
+        <BackIcon />
+        Back
+      </KButton>
+      <KButton @click="handleModalClose">
+      Proceed
+      <ForwardIcon />
+      </KButton>
     </template>
-    <template v-slot:action-buttons>
-      <KButton appearance="tertiary" class="horizontal-spacing" @click="slottedIsOpen3 = false">Pass</KButton>
-      <KButton appearance="primary" @click="slottedIsOpen3 = false">I sure do!</KButton>
+  </KModal>
+</template>
+
+<script setup lang="ts">
+const modalVisible = ref<boolean>(false)
+
+const handleModalClose = () => {
+  modalVisible.value = false
+}
+</script>
+```
+
+:::tip NOTE
+KModal takes care of placement and spacing between the buttons when using `footer-actions` slot. However, should you want to mimic that container behaviour using, for example, [`footer`](#footer) slot, we recomend you use `$kui-space-40` [design token](https://github.com/Kong/design-tokens) for spacing between the buttons and place the container containing the buttons on the right using `margin-left: $kui-space-auto;` rule.
+:::
+
+### content
+
+By default KModal provides you with the standard layout: modal header (optional), content section and footer. However, should you need to make a custom modal window, you can use `content` slot for your content. It will provide you with just a padded container with white background.
+
+:::warning NOTE
+Focus-trap requires at least one tabbable element to be present in modal at all times. You can learn more about what elements are considered tabbable [here](https://github.com/focus-trap/tabbable).
```

Comment:
Can you provide guidance here on what this means? e.g. ensure your modal has at least one element ... blah